src/main/java/seedu/address/model/ingredient/UniqueIngredientList.java

```diff
@@ -96,48 +96,65 @@ public Ingredient find(IngredientName ingredientName) throws IngredientNotFoundE
     }
 
     /**
-     * Stocks up the ingredients which are keys in HashMap {@code recipe} by increasing the number of units of each
-     * ingredient.
+     * Stocks up the ingredients which are keys in HashMap {@code requiredIngredients} by increasing the number
+     * of units of each ingredient.
      * @throws  IngredientNotFoundException if the ingredient does not exist in the list.
      */
-    public void stockUp(HashMap<IngredientName, Integer> recipe) throws IngredientNotFoundException {
-        requireNonNull(recipe);
-        for (HashMap.Entry<IngredientName, Integer> ingredientPair : recipe.entrySet()) {
+    public void stockUp(HashMap<IngredientName, Integer> requiredIngredients) throws IngredientNotFoundException {
```

Comment:
The project use Interface as method parameter
`public Person(Name name, Phone phone, Email email, Address address, Remark remark, Set<Tag> tags) {`
I think maybe it is better to take Map as the parameter.
```suggestion
    public void stockUp(Map<IngredientName, Integer> requiredIngredients) throws IngredientNotFoundException {
```

Reply:
I agree w Yi Can on this. Using interface here ensures minimal code changes in the event the implementation changes. Also, you can easily switch between the implementation of Map, should you wish to.

Reply:
Sounds logical, okay will make the change 👍